### PR TITLE
Test for memory check using md5sum

### PR DESCRIPTION
**Pull request by harish-24**
Test creates a chunk of RAM size and checks its md5sum multiple times so that memory integrity is intact

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: memory/sum_check.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            mdsum.append(process.system_output('md5sum %s' % self.ddfile))
+            self.log.info("MD5 : %s", mdsum[i - 1])
+
+        if len(set(mdsum)) > 1:
+            self.fail('Md5sum for created file differs')
+
+    def tearDown(self):
+        process.system("echo 3 > /proc/sys/vm/drop_caches",
+                       shell=True, ignore_status=True)
+        os.remove(self.ddfile)
```
Comment (narasimhan-v) [inline memory/sum_check.py]: Doesn't the self.workdir get deleted automatically ?

File: memory/sum_check.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            self.fail("Chunk creation failed due to %s" % details)
+        for i in range(self.iter):
+            mdsum.append(process.system_output('md5sum %s' % self.ddfile))
+            self.log.info("MD5 : %s", mdsum[i - 1])
+
+        if len(set(mdsum)) > 1:
+            self.fail('Md5sum for created file differs')
+
+    def tearDown(self):
+        process.system("echo 3 > /proc/sys/vm/drop_caches",
```
Comment (narasimhan-v) [inline memory/sum_check.py]: `genio.write_file()`

File: memory/sum_check.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        self.log.info("Creating chunk with dd")
+        try:
+            process.system('dd if=/dev/urandom of=%s bs=%s count=1024' %
+                           (self.ddfile, self.memsize))
+        except process.CmdError as details:
+            self.fail("Chunk creation failed due to %s" % details)
+        for i in range(self.iter):
+            mdsum.append(process.system_output('md5sum %s' % self.ddfile))
+            self.log.info("MD5 : %s", mdsum[i - 1])
```
Comment (narasimhan-v) [inline memory/sum_check.py]: Why `i - 1` ?

File: memory/sum_check.py (hunk 1 above)
Comment (harish-24) [reply memory/sum_check.py]: Oh, missed that. Will remove.

File: memory/sum_check.py (hunk 3 above)
Comment (harish-24) [reply memory/sum_check.py]: Got confused with range(). Right, `i` would be fine.

File: memory/sum_check.py (hunk 2 above)
Comment (harish-24) [reply memory/sum_check.py]: sure, will adapt.

**Comment by harish-24** [comment]
@narasimhan-v Thanks for the review. Changes are made.

**Comment by PraveenPenguin** [comment]
@harish-24 thanks for patch , @narasimhan-v thanks for ACK

**Pull request merged**